DSCResources/MSFT_xVHD/MSFT_xVHD.psm1

```diff
@@ -75,6 +80,17 @@ function Set-TargetResource
         Throw "Please ensure that Hyper-V role is installed with its PowerShell module"
     }
 
+    # Check the module combinations make sense
+    if($ParentPath -and $Type -ne "Differencing")
```

Comment:
Could you add a unit test for this resource. It doesn't have to cover the entire thing but at least validate these parameter conditions are working properly e.g. It 'Should throw when ParentPath defined and Type -ne Diff'